packages/react-components/react-drawer/src/components/Drawer/renderDrawer.tsx

```diff
@@ -8,6 +8,17 @@ import type { DrawerState, DrawerSlots } from './Drawer.types';
 export const renderDrawer_unstable = (state: DrawerState) => {
   const { slots, slotProps } = getSlots<DrawerSlots>(state);
 
-  // TODO Add additional slots in the appropriate place
+  if (state.type === 'temporary') {
+    return slots.dialog && slots.dialogSurface ? (
+      <slots.dialog {...slotProps.dialog}>
+        <slots.dialogSurface {...slotProps.dialogSurface} />
+      </slots.dialog>
+    ) : null;
+  }
+
+  if (!state.open) {
```

Comment:
can we move this at the top of the function ? we dont have to do any computation (like calling hooks etc) if we don't wanna do anything

Reply:
Not really. 
When the type is `overlay`, then we render a Dialog. If not, then we can only "render" if `open`. That's why this condition is here to prevent any renderization.